src/hooks/useNavigation.js

```diff
@@ -4,10 +4,18 @@ import { useSoftkey } from 'hooks'
 export const useNavigation = (origin, containerRef, axis, elementsSelector = '[data-selectable]') => {
   const [current, setCurrent] = useState({ type: null, index: 0, key: null })
 
-  const getAllElements = () => document.querySelectorAll(elementsSelector)
+  const getAllElements = () => {
+    if (containerRef.current === undefined) {
+      return document.querySelectorAll(elementsSelector)
+    }
+
+    const grandparentContainer = containerRef.current.parentNode.parentNode
```

Comment:
Why do you provide a containerRef that is actually 2 levels down from the node we actually need. Can you explain how this works?

Reply:
This is because we need the `input` element to also be part of the elements being returned by `getAllElements`. The `input` is also `data-selectable` and there are at least three cases where the input is found two level away from where `ref={containerRef}` is being set: [ArticleLanguage.js#L59](https://github.com/wikimedia/wikipedia-kaios/blob/master/src/components/ArticleLanguage.js#L59), [Language.js#L51](https://github.com/wikimedia/wikipedia-kaios/blob/master/src/components/Language.js#L51), [Search.js#L64](https://github.com/wikimedia/wikipedia-kaios/blob/master/src/components/Search.js#L64)

The idea then is to make the scope of the container big enough to contain all needed elements but small enough to also ignore other elements when popups are stacked. I tried a couple different approaches for this

* For example moving the `ref={containterRef}` placement on level up in RadioListView and ListView, and then querying the `containerRef` parent node (instead of grandparent). This query does include the needed `input` and makes navigation work, but it messes up the screen scrolling code which also uses `containerRef` ([useNavigation.js#L80](https://github.com/wikimedia/wikipedia-kaios/blob/master/src/hooks/useNavigation.js#L80)). Scrolling code would have to be updated too.
* Also tried setting `ref={containterRef}` actually two levels out, which means taking it outside of the RadioViewList/ListView component and into Search/Language/ArticleLanguage. But again this meant even further changes needed to get navigation to work. 

So ultimately I’m proposing to grab the grandparent right from within `getAllElements` because it seems to be the least intrusive change that gets us the working behavior, correctly supporting navigation with stacked popups. Does this make sense? Can we think of a better way?

Reply:
Thanks for the explanation. I remember now the issue with the input being conceptually part of the list for navigation but not for "scroll into view".

If there's a simple way to capture and use 2 refs, one for each purpose, I think that would be more clear but if not, a prominent comment explaining what's going on could do.

Reply:
That's a great idea, what about this: https://github.com/wikimedia/wikipedia-kaios/pull/210/commits/95900ec7965529e1e5bc95faea82b7e003ba5ac0

Reply:
yeah, it looks good now 👍